android-json-form-wizard/src/main/java/com/vijay/jsonwizard/presenters/JsonFormFragmentPresenter.java

```diff
@@ -228,67 +232,70 @@ public void validateAndWriteValues() {
         for (View childView : formFragment.getJsonApi().getFormDataViews()) {
             ValidationStatus validationStatus = validateView(childView);
             String key = (String) childView.getTag(R.id.key);
-            String openMrsEntityParent = (String) childView.getTag(R.id.openmrs_entity_parent);
-            String openMrsEntity = (String) childView.getTag(R.id.openmrs_entity);
-            String openMrsEntityId = (String) childView.getTag(R.id.openmrs_entity_id);
-            Boolean popup = (Boolean) childView.getTag(R.id.extraPopup);
-            String fieldKey = mStepName + "#" + getStepTitle() + ":" + key;
-
-            if (childView instanceof MaterialEditText) {
-                MaterialEditText editText = (MaterialEditText) childView;
-
-                String rawValue = (String) editText.getTag(R.id.raw_value);
-                if (rawValue == null) {
-                    rawValue = editText.getText().toString();
-                }
+            String address = (String) childView.getTag(R.id.address);
+            if (mStepName.equals(address.split(":")[0])) {
```

Comment:
The assumption here is that only the current step will need to be validated at any given time. 
I think when clicking save/submit maybe the whole form may need validation in some forms validation is skipped when jumping steps